frontend/src/metabase/visualizations/visualizations/Text/Text.jsx

```diff
@@ -94,10 +99,46 @@ export default class Text extends Component {
   preventDragging = e => e.stopPropagation();
 
   render() {
-    const { className, gridSize, settings, isEditing, isPreviewing } =
-      this.props;
+    const {
+      className,
+      dashboard,
+      dashcard,
+      gridSize,
+      settings,
+      isEditing,
+      isPreviewing,
+      parameterValues,
+    } = this.props;
     const isSingleRow = gridSize && gridSize.height === 1;
 
+    let parametersByTag = {};
+    if (dashcard && dashcard.parameter_mappings) {
+      parametersByTag = dashcard.parameter_mappings.reduce((acc, mapping) => {
+        const tagId = mapping.target[1];
+        const parameter = dashboard.parameters.find(
+          p => p.id === mapping.parameter_id,
+        );
+        if (parameter) {
+          const parameterValue = parameterValues[parameter.id];
+          return {
+            ...acc,
+            [tagId]: { ...parameter, value: parameterValue },
+          };
+        } else {
+          return acc;
+        }
+      }, {});
+    }
+
+    let content = settings["text"];
+    if (!_.isEmpty(parametersByTag)) {
+      // Temporarily override language to use site language, so that all viewers of a dashboard see parameter values
+      // translated the same way.
+      content = withInstanceLocalization(() => {
+        return substitute_tags(content, parametersByTag);
+      });
```

Comment:
```suggestion
      content = withInstanceLocalization(() => substitute_tags(content, parametersByTag));
```

nit: shorthand for arrow function that has only one line.